codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java

```diff
@@ -548,15 +550,92 @@ public String doubleShape(DoubleShape shape) {
 
   @Override
   public String unionShape(UnionShape shape) {
+    final ServiceShape serviceShape = context
+        .model()
+        .expectShape(context.settings().getService(context.model()).toShapeId())
+        .asServiceShape().get();
+    final String internalDafnyType = DafnyNameResolver.getDafnyType(
+      shape,
+      context.symbolProvider().toSymbol(shape)
+    );
+    writer.addImportFromModule("github.com/dafny-lang/DafnyRuntimeGo", "dafny");
+    String returnType = DafnyNameResolver.getDafnyType(
+        shape,
+        context.symbolProvider().toSymbol(shape)
+      );
+    String someWrapIfRequired = "%s(%s)";
+    if (this.isOptional) {
+      returnType = "Wrappers.Option";
+      someWrapIfRequired =
+        "Wrappers.Companion_Option_.Create_Some_(%s(%s))";
+    }
+    final String functionInit =
+      """
+      func() %s {
+          switch %s.(type) {""".formatted(returnType, dataSource);
+    StringBuilder eachMemberInUnion = new StringBuilder();
+    for (final var member : shape.getAllMembers().values()) {
+      final String memberName = context.symbolProvider().toMemberName(member);
+      final Shape targetShape = context.model().expectShape(member.getTarget());
+      final String baseType = DafnyNameResolver.getDafnyType(
+        targetShape,
+        context.symbolProvider().toSymbol(targetShape)
+      );
+      String dataSourceInput = dataSource
```

Comment:
final

Reply:
This repository does not strictly require final and it is missed on a lot of files in main like 
- https://github.com/smithy-lang/smithy-dafny/blob/main-1.x/codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydotnet/AwsSdkDotNetNameResolver.java#L254-L255
- https://github.com/smithy-lang/smithy-dafny/blob/main-1.x/codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/common/nameresolver/SmithyNameResolver.java

I have spent a lot of time finding variable to make it final. I think if adding a final to all the variable is a blocking comment then someone should add a checkstyle (like https://checkstyle.sourceforge.io/) to this repo that will automatically check and maybe put final to the java code but looking in the rest of the code right now this does not look like a blocking comment.

Reply:
Done here https://github.com/smithy-lang/smithy-dafny/pull/608/commits/75c8e6a384e4c085e161578cfb682151c7d02e8e

Maybe in some more other commits